### PR TITLE
Add better support for transitions

### PR DESCRIPTION
**Pull request by peeepers**
- More granular transition timing options (selecting colors manually can be smooth)
- Initial support for automatic discovery using netdisco
- Removed automatic_add options (was redundant and had no purpose)
- Added logging library (was missing)
- Updated README outlining the new transition settings

### CODE REVIEW

**Comment by mxtra** [comment]
Thank you for your work! Maybe we should think about integrating into HA.

File: yeelight.py [hunk 1]
```diff
@@ -95,7 +117,9 @@ def turn_on(self, **kwargs):
         transtime = 0
 
         if ATTR_TRANSITION in kwargs:
-            transtime = kwargs[ATTR_TRANSITION] * 1000
```
Comment (mxtra) [inline yeelight.py]: Forgive my ignorance, but I'm not sure if this is a good idea. The light component (https://home-assistant.io/components/light/) asks for a transition time in seconds - that is the reason why I was converting it here. Please correct me if I'm wrong.

Comment (jared-jensen) [inline yeelight.py]: I saw it, too. The thing is, I reviewed the other lights that supported this feature, and many don't stick to that standard anyway (including hue).

https://github.com/home-assistant/home-assistant/blob/d7452f9d5d100d126d52dca3e46ae709fab8b339/homeassistant/components/light/osramlightify.py#L139

https://github.com/home-assistant/home-assistant/blob/08bacd8e310c03949a598f72308e474f647a5531/homeassistant/components/light/hue.py#L259

https://github.com/home-assistant/home-assistant/blob/1f1ac02457124d456f4b0251f3ec888e31474e97/homeassistant/components/light/limitlessled.py#L137

Comment (jared-jensen) [inline yeelight.py]: @mxtra agree or disagree my friend?

**Pull request merged**